read-http-headers skips unknown headers from the result

dkochmanski opened the pull request:
This merge request makes read-http-headers to not cons unknown headers into the result (instead of returning strings as a name), and updates the list of known words with the up-to-date list of known headers (taken from MDN).

dkochmanski committed:
read-http-headers: don't collect unknown headers

After c29d92e6077dbee59a18fa7b410176b0f31205d1 READ-HTTP-HEADERS stopped to
intern unknown headers, but it still collected strings instead.

The function READ-HTTP-HEADERS is specified to return pairs (name . value) where
the name is a keyword. This change is backward incompatible, case in point:
- DRAKMA had to change a predicate to EQUALP when the header is a string
- HUNCHENSOCKET failed, because alist was uses as initargs after transformation

This commit changes READ-HTTP-HEADERS to not collect unknown headers. Previous
commit added more standard keywords, and the next commit will introduce a
function that returns all headers.

Fixes #14.

dkochmanski committed:
Update +KNOWN-WORDS+ with headers mentioned in MDN

The header list may be found here:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers

I've also added a missing http request method "PATCH"

The new list of headers contains a union of old headers and mdn headers. Chunga
had the following words that are not mentioned in MDN:

("Warning" "TimeOut" "Pragma" "Overwrite" "Lock-Token" "If" "Destination"
 "Depth" "DAV" "Content-MD5" "Accept-Charset")

Moreover I've included all headers mentioned in MDN including Experimental,
Non-Standard and Deprecated.

With this list HUNCHENSOCKET successfully serves connections. See #14.

daninus14 commented:
For reference on the discussion: See issue https://github.com/edicl/chunga/issues/14

dkochmanski committed:
read-http-headers: add a new function read-raw-http-headers

This function works like read-http-headers but it doesn't ignore anything and
returns a list of pairs where both name and value are strings.

This commit includes documentation.

dkochmanski committed:
read-http-headers, read-raw-http-headers: factor out common code

dkochmanski commented:
@hanshuebner as discussed in the issue thread:
- I've removed a warning
- Introduced read-raw-http-headers
- Updated the documentation

I've also factored out a common code path for reading headers between both functions.

hanshuebner commented on read.lisp:
```diff
@@ -145,47 +145,54 @@ end.  Might return STRING."
+                   (let ((next (peek-char* stream)))
+                     (unless (whitespacep next)
+                       (return)))
+                   ;; we've seen whitespace starting a continutation,
```
```suggestion
                   ;; we've seen whitespace starting a continuation,
```

hanshuebner reviewed (changes requested):
Looks great, thank you!

hanshuebner committed:
Update read.lisp

dkochmanski commented:
Likewise, thanks!
